File: app/src/main/java/pt/isec/supraindustries/tp_amov/oldfiles/Produto.java
package pt.isec.supraindustries.tp_amov.oldfiles;

import pt.isec.supraindustries.tp_amov.Data.Unidade;

public class Produto {
    String nome;
    String marca;
    float preco;
    int quantidade;
    Unidade unidade;
    boolean comprado;
    String notas;



    public Produto(String nome, String marca, int quantidade) {
        this.nome = nome;
        this.marca = marca;
    }
}

File: app/src/main/java/pt/isec/supraindustries/tp_amov/Data/Produto.kt
package pt.isec.supraindustries.tp_amov.Data

import java.io.Serializable

data class Produto(var nome: String, var marca: String="N/A", var unidade: Unidade?=null, var categoria: Categoria?=null, var notas: String = "" ) : Serializable {
    var precos: ArrayList<Float> = ArrayList<Float>(10)

    override fun toString(): String {
        var retString : String
        retString = nome +" | " + marca

        if(categoria != null){
            retString +=  " " +categoria!!.nome
        }
        return retString
    }

    fun addPreco(preco : Float){
        if(precos.size<10){
            precos.add(0,preco)
            return
        }
        precos.removeAt(9)
        precos.add(0,preco)
    }

    fun remPreco(index: Int){
        precos.removeAt(index)
    }
}